plugin/src/main/java/org/opensearch/ml/model/MLModelManager.java

```diff
@@ -388,6 +450,95 @@ public void registerMLModel(MLRegisterModelInput registerModelInput, MLTask mlTa
         }
     }
 
+    private UpdateRequest createUpdateModelGroupRequest(
+        Map<String, Object> modelGroupSourceMap,
+        String modelGroupId,
+        long seqNo,
+        long primaryTerm,
+        int updatedVersion
+    ) {
+        modelGroupSourceMap.put(MLModelGroup.LATEST_VERSION_FIELD, updatedVersion);
+        modelGroupSourceMap.put(MLModelGroup.LAST_UPDATED_TIME_FIELD, Instant.now().toEpochMilli());
+        UpdateRequest updateModelGroupRequest = new UpdateRequest();
+
+        updateModelGroupRequest
+            .index(ML_MODEL_GROUP_INDEX)
+            .id(modelGroupId)
+            .setIfSeqNo(seqNo)
+            .setIfPrimaryTerm(primaryTerm)
+            .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
+            .doc(modelGroupSourceMap);
+
+        return updateModelGroupRequest;
+    }
+
+    private int incrementLatestVersion(Map<String, Object> modelGroupSourceMap) {
+        return (int) modelGroupSourceMap.get(MLModelGroup.LATEST_VERSION_FIELD) + 1;
+    }
+
+    private void indexRemoteModel(
```

Comment:
Do we need to the other method: `private void indexRemoteModel(MLRegisterModelInput registerModelInput, MLTask mlTask, String modelVersion)` as we are invoking this method only.

Reply:
Yes it's needed for registering non-remote models. Actually if you use the other method, the model_id will not be included and the model registering could be asynchronize.